src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+    private static final int FIGURES_COUNT = 5;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MIN_SIZE = 1;
+    private static final int MAX_SIZE = 10;
```

Comment:
1. Services we rely on are usually private final
2. Also services are not immutable classes so technically they cannot be constants even when private static final
```suggestion
    private static final int FIGURES_COUNT = 5;
    private static final int DEFAULT_RADIUS = 10;
    private static final int MIN_SIZE = 1;
    private static final int MAX_SIZE = 10;
    
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```

Reply:
Will this not lead to redundancy? Not in this case, but in general. One Random object is enough for any number of created FigureSupplier. I did not proceed from immutability, but from the fact that the constructor will be called only once.